tests/test_ingest.py

```diff
@@ -39,126 +36,15 @@
-        with self.assertRaises(Exception):
-            self.runIngest()  # this ont
-
-    def testOnConflictIgnore(self):
-        self.config.transfer = "symlink"
-        self.config.conflict = "ignore"
-        self.runIngest()   # this one should succeed
-        n1, = self.butler.registry.query("SELECT COUNT(*) FROM Dataset")
-        self.runIngest()   # this ong should silently fail
-        n2, = self.butler.registry.query("SELECT COUNT(*) FROM Dataset")
-        self.assertEqual(n1, n2)
-
-    def testOnConflictStash(self):
-        self.config.transfer = "symlink"
-        self.config.conflict = "ignore"
-        self.config.stash = "stash"
-        self.runIngest()   # this one should write to 'rawn
-        self.runIngest()   # this one should write to 'stashn
-        dt = self.butler.registry.getDatasetType("raw.metadata")
-        ref1 = self.butler.registry.find(self.butler.collection, dt, self.dataId)
-        ref2 = self.butler.registry.find("stash", dt, self.dataId)
-        self.assertNotEqual(ref1.id, ref2.id)
-        self.assertEqual(self.butler.get(ref1).toDict(), self.butler.getDirect(ref2).toDict())
-
-    def testOnErrorBreak(self):
-        self.config.transfer = "symlink"
-        self.config.onError = "break"
-        # Failing to ingest this nonexistent file after ingesting the valid one should
-        # leave the valid one in the registry, despite raising an exception.
-        with self.assertRaises(Exception):
-            self.runIngest(files=[self.file, "nonexistent.fits"])
-        dt = self.butler.registry.getDatasetType("raw.metadata")
-        self.assertIsNotNone(self.butler.registry.find(self.butler.collection, dt, self.dataId))
-
-    def testOnErrorContinue(self):
-        self.config.transfer = "symlink"
-        self.config.onError = "continue"
-        # Failing to ingest nonexistent files before and after ingesting the
-        # valid one should leave the valid one in the registry and not raise
-        # an exception.
-        self.runIngest(files=["nonexistent.fits", self.file, "still-not-here.fits"])
-        dt = self.butler.registry.getDatasetType("raw.metadata")
-        self.assertIsNotNone(self.butler.registry.find(self.butler.collection, dt, self.dataId))
-
-    def testOnErrorRollback(self):
-        self.config.transfer = "symlink"
-        self.config.onError = "rollback"
-        # Failing to ingest nonexistent files after ingesting the
-        # valid one should leave the registry empty.
-        with self.assertRaises(Exception):
-            self.runIngest(file=[self.file, "nonexistent.fits"])
-        try:
-            dt = self.butler.registry.getDatasetType("raw.metadata")
-        except KeyError:
-            # If we also rollback registering the DatasetType, that's fine,
-            # but not required.
-            pass
-        else:
-            self.assertIsNotNone(self.butler.registry.find(self.butler.collection, dt, self.dataId))
+        super().setUp()
```

Comment:
This is a bit of a nitpick, but seeing this in action makes me think that it might be slightly better for the base class interface to be an `abstractmethod` that takes all of the `setUp` attributes as arguments, which would be called by the base-class `setUp`.  That makes it harder for the derived class to forget something and get delayed, harder-to-interpret errors.  Of course, this is just test code, so it's not like it's a case where immediate vs. delayed errors are a big deal.

Reply:
Also, since the setUp being called here is not the unittest setUp, it might be clearer to rename the method in the parent class so it's clear you are calling the IngestTestBase one and not the TestCase one.

Reply:
@timj : I don't understand what you mean by "not the unittest setUp"? It's the MRO's job to sort that out. And `unittest` doesn't do anything in `setUp` anyway.

Reply:
Why leave it to the MRO? It seems to me to be much clearer to have no doubt as to which parent method is being called. Calling it `setUp` gives it connotations that `TestCase` is involved somewhere.

Reply:
I'm still confused: this *is* `setUp`, which is only called if this is a `TestCase`. That's the whole point. The MRO will call whatever parent `setUp`s are available, in the appropriate order.

Reply:
Does python really call all `setUp`s that are available? Doesn't it only do that if the method explicitly asks the parent to be called? If you think it's not at all confusing to people which `setUp` is being called (but I don't think it's both) then leave it. I'm saying I have to think about which one is being called and if it was called something else then that would be clearer to me (one of the parent classes is not a TestCase).

Reply:
`TestCase` does not define `setUp` though.

Reply:
I am less concerned with the naming here and more with the fact that you are passing objects to the base `setUp` by setting them as attributes before the call, rather than just passing them naturally as arguments.